### ISSUE TITLE
Fix the blocks aria-label

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#5505]
Previously, the blocks container used to be focusable with a `tabindex="0"` and it was labeled with an `aria-label` attribute to give each block a name. See https://github.com/WordPress/gutenberg/commit/1a17201bae29343b909367852f21a188f19d7d16 and related #1618

This was the related markup:

![screen shot 2017-06-30 at 17 55 40](https://user-images.githubusercontent.com/1682452/27744056-6da56ed8-5dbe-11e7-88c2-5096ddf04337.png)

This was a nice feature. When navigating through the blocks using assistive technologies, and landing on a block, each block announced its type, for example: 
`Block: Paragraph`
`Block: Image`
etc.

After recent changes, this aria-label is now on a non-focusable div, see https://github.com/WordPress/gutenberg/blob/8f07c807542fc25688d34537d2fb16f6f8dbb8c0/editor/components/block-list/block.js#L603

This is not correct and should be fixed. The aria-label should be set on a container that actually receives focus. It also needs an explicit `role` attribute, e.g. `role="group"` if there are no other roles that suit better.

Regardless, it's not clear to me where the focus currently lands and if there are other pending changes planned for the blocks interaction.

Worth also considering the pending experiment about "navigation mode / edit mode":
https://github.com/WordPress/gutenberg/pull/3195
https://github.com/WordPress/gutenberg/issues/2031

The final goal should always be:
- allow keyboard and assistive technologies users to easily navigate through blocks
- correctly label each block

**Comment by IreneStr** [comment]
I've opened a PR to fix the aria-label: https://github.com/WordPress/gutenberg/pull/7305
The role-part of this issue is still open and will need some investigation about what the most suitable role.

**Comment by jorgefilipecosta** [comment]
Hi @afercia, the PR https://github.com/WordPress/gutenberg/pull/7305 should have fixed the issue. If you feel any other improvement related to this issue is required to feel free to reopen the issue.
Thank you @afercia and @IreneStr for fixing this problem.

**Comment by afercia** [comment]
Hi @jorgefilipecosta thanks and thanks @IreneStr 👍
